clients/application-kafka/src/main/java/org/eclipse/hono/application/client/kafka/impl/KafkaBasedCommandSender.java

```diff
@@ -375,26 +378,24 @@ void setCorrelationIdSupplier(final Supplier<String> correlationIdSupplier) {
          * Starts a timer so that after the given timeout value, this promise shall get failed if not completed already.
          *
          * @param correlationId The identifier to use for correlating a command with its response.
-         * @param timeout The timeout duration to use for the timer.
+         * @param timeoutInMs The timeout duration in milliseconds to use for the timer.
          * @param timeOutHandler The operation to run after this promise got failed as part of a timeout.
          * @param span The active OpenTracing span for this operation. It is not to be closed in this method!
          *             An implementation should log (error) events on this span and it may set tags and use this span as
          *             the parent for any spans created in this method.
-         * @throws NullPointerException if the timeout or span is {@code null}.
+         * @throws NullPointerException if the span is {@code null}.
          */
-        ExpiringCommandPromise(final String correlationId, final Duration timeout, final Handler<Void> timeOutHandler,
+        ExpiringCommandPromise(final String correlationId, final long timeoutInMs, final Handler<Void> timeOutHandler,
                 final Span span) {
-            Objects.requireNonNull(timeout);
             Objects.requireNonNull(span);
 
             this.span = span;
-            timerId = vertx.setTimer(timeout.toMillis(), id -> {
-                final long timeOutInMs = timeout.toMillis();
+            timerId = vertx.setTimer(timeoutInMs, id -> {
```

Comment:
This will cause an `IllegalArgumentException` if the timeout is configured as `0`. 
For the `RequestResponseClient#setRequestTimeout` we also allow `0`, meaning no timeout, so I think there should be no exception here in that case.

Reply:
@calohmn Also `RequestResponseClient#setRequestTimeout` doesn't allow negative values. The `Duration` class accepts negative values too. I have also introduced validation to check if the timeout is not negative in par with the `RequestResponseClient#setRequestTimeout`. I have pushed a new commit for it. Would you mind taking a look?